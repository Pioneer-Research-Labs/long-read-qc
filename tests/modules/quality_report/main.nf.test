nextflow_process {

    name "Test Process quality_report"
    script "modules/quality_report/main.nf"
    process "quality_report"

    test("Should generate fastplong html") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.fromList([ tuple([ id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file("${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/5HKJT6_1_sample_1.fastq.gz"),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/c.00323.dna'))])
                """
            }
        }

        then {
            assert process.success
            assert process.out[0][0].size() == 3
            assert snapshot("$launchDir/results/Gateway_EmptyBarcodedLib1_20241217/fastplong.html").match()
        }

    }

}
