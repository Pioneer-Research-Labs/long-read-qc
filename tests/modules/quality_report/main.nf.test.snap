{
    "Should generate fastplong html": {
        "content": [
            "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/4dcc26ddd7a1138d9fa98273f641f49e/results/Gateway_EmptyBarcodedLib1_20241217/fastplong.html"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-31T15:24:30.771898"
    }
}